src/rdkafka.h

```diff
@@ -6904,6 +6926,58 @@ rd_kafka_group_result_name(const rd_kafka_group_result_t *groupres);
 RD_EXPORT const rd_kafka_topic_partition_list_t *
 rd_kafka_group_result_partitions(const rd_kafka_group_result_t *groupres);
 
+/**
+ * @brief Topic Partition Result provides per-topic+partition operation result
+ *
+ */
+
+/**
+ * @returns the topic name from the topic partition result object.
+ * @remark lifetime of the returned string is the same as the \p
+ *         partition_result.
+ */
+RD_EXPORT const char *rd_kafka_topic_partition_result_topic(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @returns the partition number from the topic partition result object.
+ * @remark lifetime of the returned integer is the same as the \p
+ *         partition_result.
+ */
+RD_EXPORT int32_t rd_kafka_topic_partition_result_partition(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @returns the error code from the topic partition result object.
+ * @remark lifetime of the returned error is the same as the \p
+ *         partition_result.
+ */
+RD_EXPORT rd_kafka_resp_err_t rd_kafka_topic_partition_result_error(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @returns the error string or NULL if there was no error.
+ * @remark lifetime of the returned string is the same as the \p
+ * partition_result.
+ */
+RD_EXPORT const char *rd_kafka_topic_partition_result_error_string(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @brief Destroy and free a topic partition result object.
+ */
+RD_EXPORT void rd_kafka_topic_partition_result_destroy(
+    rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @brief Destroy and free an array of topic partition result
+ *        objects.
+ * @remark The array itself is not freed.
+ */
+RD_EXPORT void rd_kafka_topic_partition_result_destroy_array(
+    rd_kafka_topic_partition_result_t **partition_results,
+    int32_t partition_results_cnt);
+
```

Comment:
I don't think we need to expose these destroy methods to public. Remove these functions.